src/reduce.ts

```diff
@@ -114,48 +115,42 @@ function reduce<T extends readonly [], Acc>(
   iterable: T,
 ): Acc;
 
-function reduce<T>(f: (acc: T, value: T) => T, iterable: Iterable<T>): T;
+function reduce<T>(f: SyncReducer<T, T>, iterable: Iterable<T>): T;
 
-function reduce<T, Acc>(
-  f: (acc: Acc, value: T) => Acc,
-  iterable: Iterable<T>,
-): Acc;
+function reduce<T, Acc>(f: SyncReducer<Acc, T>, iterable: Iterable<T>): Acc;
 
 function reduce<T, Acc>(
-  f: (acc: Acc, value: T) => Acc,
+  f: SyncReducer<Acc, T>,
   seed: Acc,
   iterable: Iterable<T>,
 ): Acc;
 
 function reduce<T>(
-  f: (acc: T, value: T) => T,
+  f: SyncReducer<T, T>,
   iterable: AsyncIterable<T>,
 ): Promise<T>;
 
 function reduce<T, Acc>(
-  f: (acc: Acc, value: T) => Acc | Promise<Acc>,
+  f: AsyncReducer<Acc, T>,
   seed: Acc | Promise<Acc>,
   iterable: AsyncIterable<T>,
 ): Promise<Acc>;
 
 function reduce<T, Acc>(
-  f: (acc: Acc, value: T) => Acc | Promise<Acc>,
+  f: AsyncReducer<Acc, T>,
   iterable: AsyncIterable<T>,
 ): Promise<Acc>;
 
 function reduce<T extends Iterable<unknown> | AsyncIterable<unknown>>(
-  f: (
-    acc: IterableInfer<T>,
-    value: IterableInfer<T>,
-  ) => IterableInfer<T> | Promise<IterableInfer<T>>,
-): (iterable: T) => ReturnValueType<T, IterableInfer<T>>;
+  f: AsyncReducer<IterableInfer<T>, IterableInfer<T>>,
+): (iterable: T) => ReturnValueType<T>;
```

Comment:
`ReturnValueType<T>` is equivalent to `ReturnValueType<T, IterableInfer<T>>`. See `ReturnValueType.ts`.